src/components/utils/aggrid/custom-set-filter.tsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, {
+    forwardRef,
+    useEffect,
+    useImperativeHandle,
+    useState,
+} from 'react';
+import { IDoesFilterPassParams, IFilterParams } from 'ag-grid-community';
+import { Option } from 'components/results/shortcircuit/shortcircuit-analysis-result.type';
+
+// TODO temporary, remove it when SCA results will use full custom headers and filters
+export default forwardRef((props: IFilterParams, ref) => {
+    const [selectedValues, setSelectedValues] = useState<string[]>(
+        props.colDef.filterParams.options.map((option: Option) => option.value)
+    );
+
+    // expose AG Grid Filter Lifecycle callbacks
+    useImperativeHandle(ref, () => {
+        return {
+            doesFilterPass(params: IDoesFilterPassParams) {
+                return true; // we filter in the back end
+            },
+
+            // meaning we dont filter if everything is selected, and empty values will remain
+            isFilterActive() {
+                return (
+                    selectedValues.length !==
+                    props.colDef.filterParams.options.length
+                );
+            },
+
+            getModel() {
+                if (!this.isFilterActive()) {
+                    return null;
+                }
+
+                return {
+                    filterType: 'text',
+                    type: 'equals',
+                    filter: selectedValues.length ? selectedValues : null, // trick to select nothing
+                };
+            },
+
+            setModel(values: string[]) {
+                setSelectedValues(values);
+            },
+        };
+    });
+
+    const onChange = (event: any) => {
+        const value = event.target.value;
+        if (selectedValues.includes(value)) {
+            setSelectedValues(selectedValues.filter((v) => v !== value));
+        } else {
+            setSelectedValues(selectedValues.concat(value));
+        }
+    };
+
+    useEffect(() => {
+        props.filterChangedCallback();
+    }, [props, selectedValues]);
+
+    return (
+        <div
+            style={{
+                display: 'flex',
+                flexDirection: 'column',
+                padding: 8,
+            }}
+        >
+            <div style={{ fontWeight: 'bold', margin: 2 }}>Select values</div>
```

Comment:
You should use MUI Box, Typography and theme spacing, but I know it's temporary...